platform/mbed_rtc_time.cpp

```diff
@@ -55,6 +55,7 @@ static time_t _rtc_lpticker_read(void)
 
 static void _rtc_lpticker_write(time_t t)
 {
+    _rtc_lp_timer->reset();
```

Comment:
If the RTC is also being used for other parts of the system, such as the low power timer, this may interfere with it. One possible less intrusive alternative is to only adjust `_rtc_lp_base` by taking into account the current time. Something like this:
```
_rtc_lp_base = t - _rtc_lp_timer->read();
```

Reply:
The current version is fine, I think. It's only resetting the internal LowPowerTimer there. 

The prefixing of everything in the file with `_rtc` is confusing.

I'd also specifically suggest not having `_rtc_lp_base` be a `uint64_t` though. It's a `time_t`. Probably wasting a word of RAM and similar amount of ROM there, at least while `time_t` is 32-bit.

Been looking into C++11 chrono recently, so getting a bit picky about correctly-typed time. Having it be a time_t makes clear that it is the actual time the clock was last adjusted, which is potentially useful information, and I could imagine an API call to read that. Russ's proposed adjustment would lose that information.